recipe/meta.yaml

```diff
@@ -22,9 +22,13 @@ source:
     patches:
       - thirdparty-mumps-linux-configure.patch  # [linux]
 
+  # CMake-based build system for Windows
+  - folder: coinor-cmake
+    url: https://github.com/traversaro/coinor-cmake/archive/0a3f9c6acd583a06206b2f1ca7160daeb3d2597f.zip
+    sha256: a63c54e8f16cb6eab4ab82e6a3dbf4be153139e3f91f999b579f52c855d8f675
```

Comment:
Bringing this repo in like this is likely to cause these files to be included in the binary. These should be downloaded and deleted before being included, I think.

Reply:
I guess this is an odd case, as most software includes their build scripts with the source code. Why are the build scripts not in the ipopt 3.13 source?

Reply:
> Bringing this repo in like this is likely to cause these files to be included in the binary. These should be downloaded and deleted before being included, I think.

Why this source folder should be included in the binary? The existing Linux and macOS binary packages does not include the Ipopt, ASL and Mumps source folder, so I don't think the `coinor-cmake` directory will end up in the binary as well.

> I guess this is an odd case, as most software includes their build scripts with the source code. Why are the build scripts not in the ipopt 3.13 source?

Ipopt ships with a autotools-based build system that while it supports MSVC, it is not friendly to use MSVC with Flang, and to link external dependencies. For this reason, both this PR and and the previous PR https://github.com/conda-forge/ipopt-feedstock/pull/34 do not use the autotools-based build system, but rather the replacement CMake-based build system maintained as a separate project in https://github.com/ycollet/coinor-cmake . As `coinor-cmake` at the moment does not support Ipopt 3.13.3, I prepare a modified version of it and I notified upstream about it https://github.com/ycollet/coinor-cmake/issues/11 . As mentioned in https://github.com/conda-forge/ipopt-feedstock/pull/47#issuecomment-72016691, the use of a separate fork instead of vendoring the build files directly in the recipe as done in https://github.com/conda-forge/ipopt-feedstock/pull/34 is to simplify exchanging patches back and forth with the upstream repo https://github.com/ycollet/coinor-cmake . Note that using a non  provided by upstream CMake-based buildsystem is exactly the strategy used for the MUMPS port in https://github.com/conda-forge/mumps-feedstock/pull/56 .

Reply:
Ok, on the source folder not being included. Thanks for checking the others.

For the ycollet build scripts and your fork, I think this is fine for now, but can you open an issue to make sure we point to the ycollet repo once your changes are merged in?

Reply:
> For the ycollet build scripts and your fork, I think this is fine for now, but can you open an issue to make sure we point to the ycollet repo once your changes are merged in?

It make sense! I opened it in https://github.com/conda-forge/ipopt-feedstock/issues/51 .